contracts/examples/WitnetRNG.sol

```diff
@@ -0,0 +1,352 @@
+        );
+    }
+
+    /// Returns `true` only when the randomness request that got posted within given block was already
+    /// reported back from the Witnet oracle, either successfully or with an error of any kind.
+    function isRandomized(uint256 _block)
+        public view
+        virtual override
+        returns (bool)
+    {
+        RandomizeData storage _data = __randomize_[_block];
+        return (
+            _data.witnetQueryId != 0 
+                && _witnetCheckResultAvailability(_data.witnetQueryId)
+        );
+    }
+
+    /// Generates a pseudo-random number uniformly distributed within the range [0 .. _range), by using 
+    /// the given `_nonce` value and the randomness returned by `getRandomnessAfter(_block)`. 
+    /// @dev Fails under same conditions as `getRandomnessAfter(uint256)` may do.
+    /// @param _range Range within which the uniformly-distributed random number will be generated.
+    /// @param _nonce Nonce value enabling multiple random numbers from the same randomness value.
+    /// @param _block Block number from which the search will start.
+    function random(uint32 _range, uint256 _nonce, uint256 _block)
+        external view
+        virtual override
+        returns (uint32)
+    {
+        return random(
+            _range,
+            _nonce,
+            keccak256(
+                abi.encode(
+                    msg.sender,
+                    getRandomnessAfter(_block)
+                )
+            )
+        );
+    }
+
+    /// Generates a pseudo-random number uniformly distributed within the range [0 .. _range), by using 
+    /// the given `_nonce` value and the given `_seed` as a source of entropy.
+    /// @param _range Range within which the uniformly-distributed random number will be generated.
+    /// @param _nonce Nonce value enabling multiple random numbers from the same randomness value.
+    /// @param _seed Seed value used as entropy source.
+    function random(uint32 _range, uint256 _nonce, bytes32 _seed)
+        public pure
+        virtual override
+        returns (uint32)
+    {
+        uint8 _flagBits = uint8(255 - _msbDeBruijn32(_range));
+        uint256 _number = uint256(
+                keccak256(
+                    abi.encode(_seed, _nonce)
+                )
+            ) & uint256(2 ** _flagBits - 1);
+        return uint32((_number * _range) >> _flagBits);
+    }
+
+    /// Requests the Witnet oracle to generate an EVM-agnostic and trustless source of randomness. 
+    /// Only one randomness request per block will be actually posted to the WRB. Should there 
+    /// already be a posted request within current block, it will try to upgrade Witnet fee of current's 
+    /// block randomness request according to current gas price. In both cases, all unused funds shall 
+    /// be transfered back to the tx sender.
+    /// @return _usedFunds Amount of funds actually used from those provided by the tx sender.
```

Comment:
```suggestion
    /// Requests the Witnet oracle to generate an EVM-agnostic and trustless randomness source.
    /// Only one randomization request per block will be actually posted to the WRB. Should there
    /// already be a posted request within current block, it will try to upgrade Witnet fee of current's
    /// block randomization request according to current gas price. In both cases, all unused funds
    /// shall be transferred back to the transaction sender.
    /// @return _usedFunds Amount of funds actually used from those provided by the tx sender.
```